### DISCUSSION TITLE
Couldnt fetch data

### DISCUSSION THREAD

**Discussion by networkandcode** [supabase/supabase#3484] [Questions]
I have an issue with the getData function, it prints the user successfully, but doesn't print resp after the await block, or the text 'user'.

```
const [ user, setUser ] = useState(supabase.auth.user());

useEffect( () => {
  if(user) {
    getData();
  }
}, [user] );

const getData = async() => {
    console.log(user);
    let resp = await supabase
      .from('wrong_answers')
      .select('*');
    console.log(resp);
    console.log('user');
 }
```

The code used for the table is as follows:
```
CREATE TABLE wrong_answers (
  id bigint generated by default as identity primary key,
  user_id uuid references auth.users not null,
  wrong_answer_count int,
  inserted_at timestamp with time zone default timezone('utc'::text, now()) not null
);

alter table wrong_answers enable row level security;

create policy "Wrong answers are public." on wrong_answers for
    select using (true);
```

Thanks in advance.

**Comment by silentworks** [comment]
When you look inside of the network tab in the developer tool, do you see the request from the supabase call being sent? if this is being sent then it should contain a response body which will help you to debug this.

**Reply by networkandcode** [reply]
@silentworks 

In the fetch/xhr tab I see the following
under Headers general:
```
Request Method: GET
Status Code: 200 
```

and in response
```
[{"id":1,"user_id":<userid>,"wrong_answer_count":19,"inserted_at":"2021-10-11T03:07:39.777977+00:00"}]
```

**Comment by nedanwr** [comment]
Correct me if I'm wrong, but isn't it;

```ts
let { data: response, error } = await supabase
         .from("wrong_answers")
         .select("*");

console.log(response);
```

**Reply by networkandcode** [reply]
@nedanwr That should be ok I think, just another way of fetching data. Same result though.

**Comment by GaryAustin1** [comment]
So probably something you know but just in case, the getData is async (and returns a promise) so the code after getData will continue to run even while the await is blocking the two console.logs.   If you have other code running generating console messages your console.logs may be much further down from the original user message.
